dotCMS/src/main/java/com/dotcms/security/secret/ServiceSecrets.java

```diff
@@ -0,0 +1,80 @@
+package com.dotcms.security.secret;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ServiceSecrets implements Serializable {
+
+    private final String serviceKey;
+
+    private final Map<String,Secret> secrets;
+
+    @JsonCreator
+    private ServiceSecrets(@JsonProperty("serviceKey") final String serviceKey,
+            @JsonProperty("secrets") final Map<String, Secret> secrets) {
+        this.serviceKey = serviceKey;
+        this.secrets = null == secrets ? new HashMap<>() : secrets ;
+    }
+
+    public String getServiceKey() {
+        return serviceKey;
+    }
+
+    public Map<String, Secret> getSecrets() {
+        return secrets;
+    }
+
+    public static class Builder {
+
+        private Map<String,Secret> secretMap = new HashMap<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Private field 'secretMap' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)